messaging/app/src/main/java/com/google/firebase/quickstart/fcm/kotlin/MainActivity.kt

```diff
@@ -115,6 +147,24 @@ class MainActivity : AppCompatActivity() {
         }
     }
 
+    private fun getAndStoreRegToken(): String {
+        var token = ""
+        runBlocking {
+            token = Firebase.messaging.token.await()
+            Log.d(TAG, "getRegistrationToken(): $token")
+            // Add token and timestamp to Firestore for this user
+            val deviceToken = hashMapOf(
+                "token" to token,
+                "timestamp" to FieldValue.serverTimestamp(),
+            )
+
+            // Get user ID from Firebase Auth or your own server
+            Firebase.firestore.collection("fcmTokens").document("myuserid")
+                .set(deviceToken)
```

Comment:
Let's await for this too
```suggestion
                .set(deviceToken).await()
```